### PR TITLE
[enhance](cooldown)accelerate cooldown task produce efficiency

### PR DESCRIPTION
**Pull request by ByteYue**
# Proposed changes

Issue Number: close #xxx
The former logic of cooldown task production is limited to maximum 2 per data dir, we could accelerate the efficiency by first removing this limitation. 
Besides the oldest_write_timestamp seems useless, it cloud be removed. 
The tablet would be assigned one cooldown priority due the cooldowntime and it's file size, the currently priority mechanism is simple, we can use one more fine-grained one in the future.
## Problem summary

Describe your changes.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [x] No
    - [ ] I don't know
2. Has unit tests been added:
    - [ ] Yes
    - [x] No
    - [ ] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [x] No
    - [ ] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [x] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [x] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 35.51 seconds
 load time: 482 seconds
 storage size: 17119322801 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230120093839_clickbench_pr_84623.html

File: be/src/olap/tablet_manager.cpp [hunk 1]
```diff
@@ -1281,39 +1281,44 @@ struct SortCtx {
...
-
-        return a.file_size > b.file_size;
-    });
+        std::for_each(
+                tablets_shard.tablet_map.begin(), tablets_shard.tablet_map.end(),
+                [&candidates](auto& tablet_pair) { candidates.emplace_back(tablet_pair.second); });
+    }
+    std::for_each(candidates.begin(), candidates.end(),
+                  [&sort_ctx_vec, &cooldown_pending_predicate](std::weak_ptr<Tablet>& t) {
+                      if (UNLIKELY(t.expired())) {
```
Comment (platoneko) [inline be/src/olap/tablet_manager.cpp:1308]: weak_ptr should be used in this way:
```
auto tablet = t.lock();
if (tablet == nullptr) {
    return;
}
```

File: be/src/olap/tablet_manager.cpp [hunk 2]
```diff
@@ -1281,39 +1281,44 @@ struct SortCtx {
...
+                          return;
+                      }
+                      const TabletSharedPtr& tablet = t.lock();
+                      std::shared_lock rdlock(tablet->get_header_lock());
+                      int64_t cooldown_timestamp = -1;
+                      size_t file_size = -1;
+                      if (cooldown_pending_predicate(tablet) ||
+                          tablet->need_cooldown(&cooldown_timestamp, &file_size)) {
```
Comment (platoneko) [inline be/src/olap/tablet_manager.cpp:1316]: `||` seems wrong

**Review by dataroaring — DISMISSED** [review]
LGTM

**Review by dataroaring — APPROVED** [review]
LGTM

**Pull request merged**